incubator/hnc/internal/reconcilers/hierarchy_config.go

```diff
@@ -398,6 +404,9 @@ func (r *HierarchyConfigReconciler) syncLabel(log logr.Logger, nsInst *corev1.Na
 		anc = anc.Parent()
 		depth++
 	}
+
+	labels[propagatePrefix+nsInst.Name] = "true"
```

Comment:
I only saw `propagate.hnc.x-k8s.io/all=true` or `propagate.hnc.x-k8s.io/none=true` annotations in the [design doc](https://docs.google.com/document/d/17J8icBEDvLLoPT4kQ4ArZcCerRweDY-TpJ48DJKpHJ0/edit#). What does `propagate.hnc.x-k8s.io/current_ns_name=true` do?

Reply:
I found out that I had to add the `propagate.hnc.x-k8s.io/current_ns_name` label manually for each ns, otherwise the selector can't identify whether to propagate or not. The `true` value is just to say that this ns has the name `current_ns_name`

Reply:
IIUC the selector will look for `current_ns_name.tree.hnc.x-k8s.io/depth: 0` label instead of a new `propagate.hnc.x-k8s.io/current_ns_name: true` label. WDYT? @adrianludwin

Reply:
Label design needs to be done _very_ carefully. @GinnyJI , why won't the depth labels that Yiqi mentioned work?

Reply:
As discussed offline, the test is updated and this label is removed